mne/preprocessing/_beer_lambert_law.py

```diff
@@ -0,0 +1,133 @@
+        iEL = np.linalg.pinv(EL)
+
+        raw._data[[ii, ii + 1]] = (raw._data[[ii, ii + 1]].T * iEL.T).T * 1e-3
+
+        raw.info['chs'][ii]['coil_type'] = FIFF.FIFFV_COIL_FNIRS_HBO
+        raw.info['chs'][ii + 1]['coil_type'] = FIFF.FIFFV_COIL_FNIRS_HBR
+
+        rename_channels(raw.info, {raw.info['chs'][ii]['ch_name']:
+                                   raw.info['chs'][ii]['ch_name'][:-4] +
+                                   ' hbo'})
+        rename_channels(raw.info, {raw.info['chs'][ii + 1]['ch_name']:
+                                   raw.info['chs'][ii + 1]['ch_name'][:-4] +
+                                   ' hbr'})
+
+    return raw
+
+
+def _channel_frequencies(raw):
+    """Return the light frequency for each channel."""
+    picks = _picks_to_idx(raw.info, 'fnirs_od')
+    freqs = np.empty(picks.size, int)
+    for ii in picks:
+        ch_name_info = re.match(r'S(\d+)-D(\d+) (\d+)',
+                                raw.info['chs'][ii]['ch_name'])
+        freqs[ii] = ch_name_info.groups()[2]
+    return freqs
+
+
+def _check_channels_ordered(raw, freqs):
+    """Check channels followed expected fNIRS format."""
+    # Every second channel should be same SD pair
+    # and have the specified light frequencies.
+    picks = _picks_to_idx(raw.info, 'fnirs_od')
+    for ii in picks[::2]:
+        ch1_name_info = re.match(r'S(\d+)-D(\d+) (\d+)',
+                                 raw.info['chs'][ii]['ch_name'])
+        ch2_name_info = re.match(r'S(\d+)-D(\d+) (\d+)',
+                                 raw.info['chs'][ii + 1]['ch_name'])
+
+        if (ch1_name_info.groups()[0] != ch2_name_info.groups()[0]) or \
+           (ch1_name_info.groups()[1] != ch2_name_info.groups()[1]) or \
+           (int(ch1_name_info.groups()[2]) != freqs[0]) or \
+           (int(ch2_name_info.groups()[2]) != freqs[1]):
+            raise RuntimeError('NIRS channels not ordered correctly')
+
+    return picks
+
+
+def _load_absorption(freqs):
+    """Load molar extinction coefficients."""
+    # Data from https://omlc.org/spectra/hemoglobin/summary.html
+    # The text was copied to a text file. The text before and
+    # after the table was deleted. The the following was run in
+    # matlab
+    # extinct_coef=importdata('extinction_coef.txt')
+    # save('extinction_coef.mat', 'extinct_coef')
+    #
+    # Returns data as [[HbO2(freq1), HbO2(freq1),
+    #                  [Hb(freq2)], Hb(freq2)]]
+    from scipy.io import loadmat
+    from scipy.interpolate import interp1d
+
+    extinction_fname = op.join(op.dirname(__file__), '..', 'data',
+                               'extinction_coef.mat')
+    a = loadmat(extinction_fname)['extinct_coef']
+
+    interp_hbo = interp1d(a[:, 0], a[:, 1], kind='linear')
+    interp_hb = interp1d(a[:, 0], a[:, 2], kind='linear')
+
+    ext_coef = np.matrix([[interp_hbo(freqs[0]), interp_hb(freqs[0])],
+                          [interp_hbo(freqs[1]), interp_hb(freqs[1])]])
```

Comment:
`numpy.matrix` is more or less deprecated, better to do everything with arrays and explicit `@` op or `np.dot`